tests/test_setup.py

```diff
@@ -524,7 +540,30 @@ async def mock_callback(hass, component):
 
 async def test_setup_import_blows_up(hass):
     """Test that we handle it correctly when importing integration blows up."""
-    with mock.patch(
+    with patch(
         "homeassistant.loader.Integration.get_component", side_effect=ValueError
     ):
         assert not await setup.async_setup_component(hass, "sun", {})
+
+
+async def test_parallel_entry_setup(hass):
+    """Test config entries are set up in parallel."""
+    MockConfigEntry(domain="comp", data={"value": 1}).add_to_hass(hass)
+    MockConfigEntry(domain="comp", data={"value": 2}).add_to_hass(hass)
+
+    calls = []
+
+    async def mock_async_setup_entry(hass, entry):
+        """Mock setting up an entry."""
+        calls.append(entry.data["value"])
+        await asyncio.sleep(0)
+        calls.append(entry.data["value"])
+        return True
+
+    mock_integration(
+        hass, MockModule("comp", async_setup_entry=mock_async_setup_entry,),
+    )
+    mock_entity_platform(hass, "config_flow.comp", None)
+    await setup.async_setup_component(hass, "comp", {})
+
+    assert calls == [1, 2, 1, 2]
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5422434)